src/pipeline.rs

```diff
@@ -0,0 +1,192 @@
+use serde::Deserialize;
+use std::{collections::HashMap, path::Path};
+use thiserror::Error;
+
+/// Data structure defining a pipeline of checks, with parameters built in
+///
+/// Rather than constructing these manually, a convenience function `load_pipelines` is provided
+/// to deserialize a set of pipelines from a directory containing TOML files defining them.
+#[derive(Debug, Deserialize, PartialEq, Clone)]
+pub struct Pipeline {
+    /// Sequence of steps in the pipeline
+    pub steps: Vec<PipelineStep>,
+    /// Minimum number of leading points required by all the tests in this pipeline
```

Comment:
Minimum or maximum? Either way, we can probably drop the adjective, "Number of extra leading data points that need to be fetched to run the tests in this pipeline" or something along the lines?

Reply:
Removed. Original intention meant minimum in the sense of "minimum `num_leading_points` a `DataCache` needs to get through this pipeline" but I can see how it would be easily misinterpreted, and I don't think it's an important detail to note in hindsight